tchannel/event.py

```diff
@@ -141,10 +150,11 @@ def register_hook(self, hook, event_type=None):
                 event_value = getattr(EventType, event_type)
                 self.register_hook(func, event_value)
 
+    @tornado.gen.coroutine
     def fire(self, event, *args, **kwargs):
         for hook in self.hooks[event]:
             try:
-                hook(*args, **kwargs)
+                yield tornado.gen.maybe_future(hook(*args, **kwargs))
```

Comment:
depending on how non-yielded performance-oriented we are: there's an `is_future` function, it might be measurably faster to only yield when necessary, since this'll mean (for our service) 3 round-trips to the IOLoop for each event (we have 3 hooks).

`is_future` is what maybe_future uses fwiw, should be reliable.

Reply:
Sounds reasonable. I'll update.